proxy/RegressionSM.cc

```diff
@@ -273,10 +275,16 @@ struct ReRegressionSM : public RegressionSM {
 
 REGRESSION_TEST(RegressionSM)(RegressionTest *t, int /* atype ATS_UNUSED */, int *pstatus)
 {
+  // create 8 state machines
+  std::vector<std::unique_ptr<ReRegressionSM>> state_machines;
+  for (auto i = 0; i < 8; i++) {
+    state_machines.push_back(make_unique<ReRegressionSM>(t));
+  }
```

Comment:
Is this the right fix? Isn't the root cause that `RegressionSM` doesn't delete it's children? At least we should have a comment here explaining what is going on and why this doesn't end up deleting state machines before they are done.